Project/Source/Components/ComponentMeshRenderer.h

```diff
@@ -24,13 +24,25 @@ class ComponentMeshRenderer : public Component {
 	void AddRenderingModeMask();
 	void DeleteRenderingModeMask();
 
+	// Dissolve
+	TESSERACT_ENGINE_API void PlayDissolveAnimation();
+	TESSERACT_ENGINE_API void ResetDissolveValues();
+
 public:
 	UID meshId = 0;
 	UID materialId = 0;
 	std::vector<float4x4> palette;
 
 	std::unordered_map<std::string, GameObject*> goBones;
 
+private:
+	void UpdateDissolveAnimation();
+
 private:
 	bool bbActive = false;
+
+	// Dissolve variables
+	float currentTime = 0.0f;
+	float dissolveThreshold = 0.0f;
+	bool dissolveAnimationFinished = true;
```

Comment:
Instead of using a dissolve duration and using UpdateDissolveAnimation to update the threshold I think it would be better to just have the threshold and edit it directly from a gameplay script. This way it would be much more configurable and easier to edit.